library/std/src/alloc.rs

```diff
@@ -353,6 +353,9 @@ fn default_alloc_error_hook(layout: Layout) {
     if unsafe { __rust_alloc_error_handler_should_panic != 0 } {
         panic!("memory allocation of {} bytes failed", layout.size());
     } else {
+        // This is the default path taken on OOM, and the only path taken on stable with std.
+        // Crucially, it does *not* call any user-defined code, and therefore there are no potential
+        // reentrancy issues. That makes it different from the default `__rdl_oom` defined in alloc.
```

Comment:
> no potential reentrancy issues.

innit more like "no potential reentrancy issues aside from the ones we foist on ourselves"?

`__rdl_oom` is an awfully descriptive name, congratulations on us for coming up with that one.

Reply:
You describe two possible differences, and either of those can independently vary, technically. You should note that `__rdl_oom` calls like `handle_alloc_error()` or whatever it actually does.

Reply:
I have tried to clarify both of these points, does that help?

Reply:
Yes, thank you!